### DISCUSSION TITLE
Triggering workflow only if Pull Request Comment

### DISCUSSION THREAD

**Discussion by mcanaves** [community/community#26833] [Actions]
<p>Hello,</p>
<p>I’m trying to run an action only when a PR comment is posted. After reading the documentation I found that the workflow should exist on the master branch so I created and merged to master but there is no way to get it.</p>
<p>This is the workflow definition:</p>
<pre><code class="lang-auto">name: Create QA environment
on:
  issue_comment:
    types: [created]
env:
  BASE_PATH: ...
  PROJECT_NAME: ...
jobs:
  wait-for-travis:
    name: Wait for Travis
    runs-on: ubuntu-latest
    if: github.event.issue.pull_request &amp;&amp; contains(github.event.comment.body, '/qa')
    outputs:
      conclusion: ${{ steps.wait-for-test.outputs.conclusion }}
    steps:
       ....
</code></pre>
<p>Any suggestions? :pray:</p>

**Comment by ghostwriter** [comment]
<pre><code class="lang-yaml">name: Create QA environment
on:
  issue_comment:
    types: [created]

jobs:
  qa-env:
    runs-on: ubuntu-latest
    outputs:
      conclusion: ${{ steps.wait-for-test.conclusion }}
    steps:
      # - name: Dump GitHub context
      #   env:
      #     GITHUB_CONTEXT: ${{ toJson(github) }}
      #   run: echo "$GITHUB_CONTEXT"

      - name: Wait for Travis
        id: wait-for-test
        if: ${{ github.event.issue.pull_request &amp;&amp; contains(github.event.comment.body, '/qa') }}
        run: echo "Event triggered!"
</code></pre>
<p>Try this workflow, it should echo <code>"Event triggered!"</code> when you comment “/qa” on a pull request.</p>

**Comment by mcanaves** [comment]
<p>Hi Nathan,</p>
<p>Thanks but I tried the workflow and it’s not triggered. Also added the dump context that should be triggered for any comment and the same result.</p>
<pre><code class="lang-yaml">name: Test action on PR comment
on:
  issue_comment:
    types: [created]

jobs:
  test-action:
    runs-on: ubuntu-latest
    steps:
      - name: Dump GitHub context
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}
        run: echo "$GITHUB_CONTEXT"
      - name: Test action on PR comment
        if: ${{ github.event.issue.pull_request &amp;&amp; contains(github.event.comment.body, '/qa') }}
        run: echo "Event triggered!"
</code></pre>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/mcanaves">@mcanaves</a> ,</p>
<p>This should work.<br>
I also set up a workflow like as below to test on my side:</p>
<pre><code class="lang-yaml">name: Issues

on:
  issue_comment:
    types: [created]

jobs:
  job1:
    runs-on: ubuntu-latest
    steps:
      - name: view the github context
        run: echo "$GITHUB_CONTEXT"
        env:
          GITHUB_CONTEXT: ${{ toJson(github) }}

  job2:
    if: ${{ github.event.issue.pull_request &amp;&amp; contains(github.event.comment.body, '/qa') }}
    runs-on: ubuntu-latest
    steps:
      - name: run for PR
        run: echo "Triggered for PR!"
</code></pre>
<p>When I added a comment (contains ‘/qa’) on a PR, <strong>job2</strong> can run normally as expected. See the <a href="https://github.com/TestWorkflowsRML/Event-Triggers/actions/runs/268423376/workflow" rel="noopener nofollow ugc">example workflow run</a>.</p>
<p>When I added a comment (contains ‘/qa’) on a issue, <strong>job2</strong> was skipped, this also is the expected result. See the <a href="https://github.com/TestWorkflowsRML/Event-Triggers/actions/runs/268424968/workflow" rel="noopener nofollow ugc">example workflow run</a>.</p>

**Comment by mcanaves** [comment, marked as answer]
<p>Finally found where was the problem, the events worked always but there was a problem with a dependant job that blocked the comment event. Thanks <a class="mention" href="/brightran">@brightran</a> and <a class="mention" href="/ghostwriter">@ghostwriter</a>.</p>
